### PR TITLE
Add drag and drop support for Windows

### PR DESCRIPTION
**Pull request by ThomasWilshaw**
As per #57 I've added drag and drop support for Windows. It seems to work okay on basic testing but I'm always a bit iffy about C pointers so please let me know if I've messed something up there.

If you drag a single file it loads correctly but if you drag several it triggers the existing code path for loading multiple files so any changes to that should work as expected on Windows

### CODE REVIEW

File: main_win32.cpp [hunk 1]
```diff
@@ -293,6 +298,31 @@ LRESULT WINAPI WndProc(HWND hWnd, UINT msg, WPARAM wParam, LPARAM lParam)
...
+        file_list = (char**)malloc(count * sizeof(char*));
+        char temp_filename[MAX_PATH];
+
+        for(UINT i = 0; i < count; ++i)
+        {
+           if (DragQueryFileA(hDrop, i, temp_filename, MAX_PATH))
+               file_list[i] = (char*)malloc(MAX_PATH * sizeof(char));
+               strcpy_s(file_list[i],MAX_PATH, temp_filename);
+        }
+
```
Comment (meshula) [inline main_win32.cpp:316]: the guts here become slightly more complex

```
        if (DragQueryFileW(hDrop, i, temp_filename, MAX_PATH)) {
            // Determine the required buffer size for the multi-byte string
            int buffer_size = WideCharToMultiByte(CP_UTF8, 0, temp_filename, -1, NULL, 0, NULL, NULL);

            // Allocate memory for the converted file name
            file_list[i] = (char*)malloc(buffer_size * sizeof(char));

            // Convert wide character filename to UTF-8
            WideCharToMultiByte(CP_UTF8, 0, temp_filename, -1, file_list[i], buffer_size, NULL, NULL);
        }
```

File: main_win32.cpp [hunk 2]
```diff
@@ -293,6 +298,31 @@ LRESULT WINAPI WndProc(HWND hWnd, UINT msg, WPARAM wParam, LPARAM lParam)
...
         SwitchToFiber(g_mainFiber);
+        break;
+    case WM_DROPFILES:
+        HDROP hDrop = reinterpret_cast<HDROP>(wParam);
+        UINT count = DragQueryFileA(hDrop, -1, NULL, 0);
+
+        char** file_list;
+        file_list = (char**)malloc(count * sizeof(char*));
+        char temp_filename[MAX_PATH];
+
```
Comment (meshula) [inline main_win32.cpp:309]: add
```
   wchar_t temp_filename[MAX_PATH];
```

File: main_win32.cpp [hunk 3]
```diff
@@ -293,6 +298,31 @@ LRESULT WINAPI WndProc(HWND hWnd, UINT msg, WPARAM wParam, LPARAM lParam)
         // back to the main fiber allowing gui to draw a frame. After a frame is drawn
         // execution is yielded back to the message fiber.
         SwitchToFiber(g_mainFiber);
+        break;
+    case WM_DROPFILES:
+        HDROP hDrop = reinterpret_cast<HDROP>(wParam);
+        UINT count = DragQueryFileA(hDrop, -1, NULL, 0);
```
Comment (meshula) [inline main_win32.cpp:304]: could we switch this to W please, and the char** to wchar_t** as necessary? I haven't check the rest of main to find out if there's more ANSI to WIDE conversion necessary, but we could start here.

**Review by meshula — CHANGES_REQUESTED** [review]
This is very cool, thanks :) Small request though.

File: main_win32.cpp (hunk 3 above)
Comment (darbyjohnston) [reply main_win32.cpp:304]: Both `src/opentimelineio/serialization.cpp` and `src/opentimelineio/deserialization.cpp` use `MultiByteToWideChar()` for opening files.

Comment (ThomasWilshaw) [reply main_win32.cpp:304]: Am I right in thinking `file_list` should remain as `char` as we're converting UTF characters in `temp_filename` into `char`s with `WideCharToMultiByte`?

Comment (meshula) [reply main_win32.cpp:305]: Yes, that's right.

**Comment by ThomasWilshaw** [comment]
I think that's all your comments resolved, let me know if I missed anything

**Review by meshula — APPROVED** [review]
lgtm!! Thanks :)

**Comment by jminor** [comment]
Thanks @ThomasWilshaw !

**Pull request merged**